### PR TITLE
Fixed pir fitness when exporting model and comm init when model inferring

### PR DESCRIPTION
**Pull request by Function-Samuel**
### PR Category
Inference

### PR Types
Bug fixes

### Description
修复了模型导出时的pir适配与推理时的通信初始化
PCard-85979

### CODE REVIEW

**Review by zhiqiu — DISMISSED** [review]
LGTM

File: paddle/fluid/framework/new_executor/instruction/instruction_util.cc [hunk 1]
```diff
@@ -153,7 +170,10 @@ phi::DeviceContext* ParseDeviceContext(pir::Operation* op,
             op_name.compare(paddle::dialect::AllReduce_Op::name()) == 0 ||
             op_name.compare(paddle::dialect::Broadcast_Op::name()) == 0 ||
             op_name.compare(paddle::dialect::BroadcastOp::name()) == 0 ||
-            op_name.compare(paddle::dialect::AllGatherOp::name()) == 0) {
+            op_name.compare(paddle::dialect::AllGatherOp::name()) == 0 ||
+            op_name.compare(paddle::dialect::MpAllreduceSum_Op::name()) == 0 ||
+            op_name.compare(paddle::dialect::CIdentity_Op::name()) == 0 ||
+            op_name.compare(paddle::dialect::CConcatOp::name()) == 0) {
```
Comment (liym27) [inline paddle/fluid/framework/new_executor/instruction/instruction_util.cc:176]: Can we not add this if condition(L168-L176) anymore, as most operators have already been migrated.

File: python/paddle/distributed/communication/stream/all_reduce.py [hunk 2]
```diff
@@ -79,15 +79,14 @@ def _all_reduce_in_static_mode(
         'all_reduce',
     )
 
-    op_type = _get_reduce_op(op, "allreduce")
```
Comment (liym27) [inline python/paddle/distributed/communication/stream/all_reduce.py]: can not delete `op_type = _get_reduce_op(op, "allreduce")` , `op_type` will be used later (L97)

Comment (Function-Samuel) [reply python/paddle/distributed/communication/stream/all_reduce.py]: yes, i ignore this, my fault.

File: paddle/fluid/framework/new_executor/instruction/instruction_util.cc (hunk 1 above)
Comment (Function-Samuel) [reply paddle/fluid/framework/new_executor/instruction/instruction_util.cc:176]: i have tried to remove them, but that will make infer error

**Review by liym27 — APPROVED** [review]
LGTM

**Review by zhangbo9674 — APPROVED** [review]
LGTM

**Pull request merged**